### PR TITLE
[CI Visibility] Remove stdout output from debug level in the CoverageCollector

### PR DESCRIPTION
**Pull request by tonyredondo**
## Summary of changes

This PR remove the debug logs from the collector standard output, these debug logs will be written in the log file.

## Reason for change

Currently we are writing all logs to the collector logger (standard output) (at least 2 log lines for each test), this introduces latency, and also there's a race condition in the last test because the collector logger requires a `DataCollectionContext` that is disposed at the end of the last test by the Test sdk, this PR fixes the race condition.

### CODE REVIEW

File: tracer/src/Datadog.Trace.Coverage.collector/CoverageCollector.cs [hunk 1]
```diff
@@ -48,7 +50,7 @@ public override void Initialize(XmlElement configurationElement, DataCollectionE
                 };
                 events.TestCaseStart += (sender, args) =>
                 {
-                    _logger?.Debug($"Test case start: {args.TestCaseName}");
+                    _logger?.Debug($"Test case start [{Interlocked.Increment(ref _testNumber)}]: {args.TestCaseName}");
```
Comment (albertvaka) [inline tracer/src/Datadog.Trace.Coverage.collector/CoverageCollector.cs:53]: This seems unrelated, is this supposed to be here?

**Review by albertvaka — APPROVED** [review]

**Comment by andrewlock** [comment]
## Benchmarks Report :snail:

Benchmarks for #3612 compared to [master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af): 
* All benchmarks have **the same** speed
* All benchmarks have **the same** allocations


The following thresholds were used for comparing the benchmark speeds:
* Mann–Whitney U test with statistical test for significance of **5%**
* Only results indicating a difference greater than **10%** and **0.3 ns** are considered.

Allocation changes below **0.5%** are ignored.


### Benchmark details


<details><summary><strong>Benchmarks.Trace.AgentWriterBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`WriteAndFlushEnrichedTraces`|net472|768μs|832ns|3.22μs|0.386| 0|0|3.22&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|559μs|687ns|2.66μs|0| 0|0|2.62&nbsp;KB|
|#3612|`WriteAndFlushEnrichedTraces`|net472|768μs|814ns|3.15μs|0.383| 0|0|3.22&nbsp;KB|
|#3612|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|566μs|645ns|2.5μs|0| 0|0|2.63&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AppSecBodyBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`AllCycleSimpleBody`|net472|24.8μs|134ns|771ns|0.28| 0|0|1.78&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`AllCycleSimpleBody`|netcoreapp3.1|21.5μs|15.3ns|55.3ns|0.0212| 0|0|1.65&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`AllCycleMoreComplexBody`|net472|193μs|711ns|2.66μs|2| 0|0|13.03&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`AllCycleMoreComplexBody`|netcoreapp3.1|179μs|280ns|1.09μs|0.0895| 0|0|12.11&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`BodyExtractorSimpleBody`|net472|281ns|0.173ns|0.669ns|0.0573| 0|0|361&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`BodyExtractorSimpleBody`|netcoreapp3.1|239ns|0.118ns|0.455ns|0.00373| 0|0|272&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`BodyExtractorMoreComplexBody`|net472|16.7μs|14.2ns|54.9ns|1.2| 0.0165|0|7.62&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`BodyExtractorMoreComplexBody`|netcoreapp3.1|13.2μs|3.98ns|14.9ns|0.0925| 0|0|6.75&nbsp;KB|
|#3612|`AllCycleSimpleBody`|net472|25.1μs|20.4ns|79.2ns|0.276| 0|0|1.78&nbsp;KB|
|#3612|`AllCycleSimpleBody`|netcoreapp3.1|23.8μs|44.9ns|174ns|0.0118| 0|0|1.65&nbsp;KB|
|#3612|`AllCycleMoreComplexBody`|net472|194μs|46.6ns|174ns|2.02| 0|0|13.03&nbsp;KB|
|#3612|`AllCycleMoreComplexBody`|netcoreapp3.1|179μs|753ns|2.82μs|0.0915| 0|0|12.11&nbsp;KB|
|#3612|`BodyExtractorSimpleBody`|net472|311ns|0.369ns|1.33ns|0.0573| 0|0|361&nbsp;B|
|#3612|`BodyExtractorSimpleBody`|netcoreapp3.1|240ns|0.15ns|0.561ns|0.00372| 0|0|272&nbsp;B|
|#3612|`BodyExtractorMoreComplexBody`|net472|15.7μs|10.5ns|39.2ns|1.21| 0.0156|0|7.62&nbsp;KB|
|#3612|`BodyExtractorMoreComplexBody`|netcoreapp3.1|12.8μs|8.59ns|31ns|0.0892| 0|0|6.75&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AspNetCoreBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`SendRequest`|net472|0ns|0ns|0ns|0| 0|0|0&nbsp;b|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`SendRequest`|netcoreapp3.1|271μs|1.55μs|11.7μs|0.253| 0|0|20.96&nbsp;KB|
|#3612|`SendRequest`|net472|0ns|0ns|0ns|0| 0|0|0&nbsp;b|
|#3612|`SendRequest`|netcoreapp3.1|263μs|1.51μs|11μs|0.245| 0|0|20.96&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.DbCommandBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`ExecuteNonQuery`|net472|1.93μs|2.69ns|10.4ns|0.162| 0.000961|0|1.03&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`ExecuteNonQuery`|netcoreapp3.1|1.46μs|1.91ns|7.39ns|0.0137| 0|0|1.02&nbsp;KB|
|#3612|`ExecuteNonQuery`|net472|1.85μs|2.93ns|11.3ns|0.163| 0.000925|0|1.03&nbsp;KB|
|#3612|`ExecuteNonQuery`|netcoreapp3.1|1.46μs|3.34ns|12.9ns|0.0141| 0|0|1.02&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.ElasticsearchBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`CallElasticsearch`|net472|2.53μs|0.996ns|3.86ns|0.195| 0|0|1.24&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`CallElasticsearch`|netcoreapp3.1|1.56μs|0.625ns|2.34ns|0.0158| 0|0|1.18&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`CallElasticsearchAsync`|net472|2.67μs|2.29ns|8.85ns|0.218| 0|0|1.37&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`CallElasticsearchAsync`|netcoreapp3.1|1.68μs|0.643ns|2.4ns|0.0173| 0|0|1.3&nbsp;KB|
|#3612|`CallElasticsearch`|net472|2.42μs|1.41ns|5.47ns|0.196| 0|0|1.24&nbsp;KB|
|#3612|`CallElasticsearch`|netcoreapp3.1|1.53μs|0.799ns|3.09ns|0.016| 0|0|1.18&nbsp;KB|
|#3612|`CallElasticsearchAsync`|net472|2.68μs|1.59ns|5.74ns|0.217| 0|0|1.37&nbsp;KB|
|#3612|`CallElasticsearchAsync`|netcoreapp3.1|1.64μs|0.752ns|2.91ns|0.0172| 0|0|1.3&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.GraphQLBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`ExecuteAsync`|net472|2.73μs|0.655ns|2.36ns|0.238| 0|0|1.5&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`ExecuteAsync`|netcoreapp3.1|1.75μs|0.696ns|2.69ns|0.0192| 0|0|1.42&nbsp;KB|
|#3612|`ExecuteAsync`|net472|2.65μs|1.69ns|6.54ns|0.238| 0|0|1.5&nbsp;KB|
|#3612|`ExecuteAsync`|netcoreapp3.1|1.73μs|1.02ns|3.81ns|0.0191| 0|0|1.42&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.HttpClientBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`SendAsync`|net472|7.29μs|1.9ns|7.36ns|0.561| 0|0|3.54&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`SendAsync`|netcoreapp3.1|4.63μs|0.785ns|2.72ns|0.044| 0|0|3.21&nbsp;KB|
|#3612|`SendAsync`|net472|7.33μs|2.48ns|9.27ns|0.56| 0|0|3.54&nbsp;KB|
|#3612|`SendAsync`|netcoreapp3.1|4.8μs|1.1ns|3.95ns|0.0431| 0|0|3.21&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.ILoggerBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`EnrichedLog`|net472|2.84μs|1.67ns|6.03ns|0.302| 0|0|1.9&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`EnrichedLog`|netcoreapp3.1|2.29μs|1.13ns|4.24ns|0.0258| 0|0|1.93&nbsp;KB|
|#3612|`EnrichedLog`|net472|2.88μs|3.17ns|11.8ns|0.302| 0|0|1.9&nbsp;KB|
|#3612|`EnrichedLog`|netcoreapp3.1|2.3μs|1.41ns|5.28ns|0.0266| 0|0|1.93&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.Log4netBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`EnrichedLog`|net472|149μs|194ns|751ns|0.74| 0.222|0|4.74&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`EnrichedLog`|netcoreapp3.1|117μs|129ns|500ns|0.0586| 0|0|4.58&nbsp;KB|
|#3612|`EnrichedLog`|net472|148μs|154ns|595ns|0.737| 0.221|0|4.74&nbsp;KB|
|#3612|`EnrichedLog`|netcoreapp3.1|119μs|140ns|525ns|0.0597| 0|0|4.58&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.NLogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`EnrichedLog`|net472|5.48μs|2.28ns|8.82ns|0.584| 0.00273|0|3.68&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`EnrichedLog`|netcoreapp3.1|4.26μs|4.83ns|18.7ns|0.0533| 0|0|4&nbsp;KB|
|#3612|`EnrichedLog`|net472|5.56μs|2.26ns|8.75ns|0.583| 0.00279|0|3.68&nbsp;KB|
|#3612|`EnrichedLog`|netcoreapp3.1|4.18μs|1.74ns|6.52ns|0.0539| 0|0|4&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.RedisBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`SendReceive`|net472|2.3μs|2.16ns|8.1ns|0.23| 0|0|1.45&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`SendReceive`|netcoreapp3.1|1.79μs|0.623ns|2.25ns|0.0188| 0|0|1.4&nbsp;KB|
|#3612|`SendReceive`|net472|2.28μs|4.55ns|17.6ns|0.23| 0|0|1.45&nbsp;KB|
|#3612|`SendReceive`|netcoreapp3.1|1.83μs|0.402ns|1.45ns|0.0193| 0|0|1.4&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SerilogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`EnrichedLog`|net472|4.76μs|1.96ns|7.61ns|0.368| 0|0|2.32&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`EnrichedLog`|netcoreapp3.1|4μs|1.08ns|4.17ns|0.0241| 0|0|1.89&nbsp;KB|
|#3612|`EnrichedLog`|net472|4.93μs|1.64ns|5.92ns|0.367| 0|0|2.32&nbsp;KB|
|#3612|`EnrichedLog`|netcoreapp3.1|4.1μs|0.854ns|3.31ns|0.0247| 0|0|1.89&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SpanBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`StartFinishSpan`|net472|1.13μs|0.317ns|1.23ns|0.141| 0|0|891&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`StartFinishSpan`|netcoreapp3.1|891ns|0.473ns|1.83ns|0.0112| 0|0|840&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`StartFinishScope`|net472|1.34μs|0.557ns|2.09ns|0.154| 0|0|971&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`StartFinishScope`|netcoreapp3.1|1.03μs|0.314ns|1.21ns|0.0129| 0|0|960&nbsp;B|
|#3612|`StartFinishSpan`|net472|1.13μs|0.731ns|2.83ns|0.141| 0|0|891&nbsp;B|
|#3612|`StartFinishSpan`|netcoreapp3.1|947ns|1.01ns|3.92ns|0.0112| 0|0|840&nbsp;B|
|#3612|`StartFinishScope`|net472|1.37μs|0.675ns|2.53ns|0.154| 0|0|971&nbsp;B|
|#3612|`StartFinishScope`|netcoreapp3.1|1.08μs|0.278ns|1.08ns|0.013| 0|0|960&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.TraceAnnotationsBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`RunOnMethodBegin`|net472|1.45μs|0.293ns|1.13ns|0.154| 0|0|971&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/a968c01d5504d892e2d6e923fd0c039548fec2af)|`RunOnMethodBegin`|netcoreapp3.1|1.21μs|0.757ns|2.93ns|0.0133| 0|0|960&nbsp;B|
|#3612|`RunOnMethodBegin`|net472|1.5μs|0.891ns|3.34ns|0.154| 0|0|971&nbsp;B|
|#3612|`RunOnMethodBegin`|netcoreapp3.1|1.09μs|0.333ns|1.25ns|0.0127| 0|0|960&nbsp;B|
</details>

File: tracer/src/Datadog.Trace.Coverage.collector/CoverageCollector.cs (hunk 1 above)
Comment (tonyredondo) [reply tracer/src/Datadog.Trace.Coverage.collector/CoverageCollector.cs:53]: it's unrelated but is a micro change to have the number of tests in the logs.

**Comment by andrewlock** [comment]
## Code Coverage Report :bar_chart:

:heavy_check_mark: Merging #3612 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/83a17a25cfd4775d64b748f0ec852d09a97f434b) will **not change** line coverage
:heavy_check_mark: Merging #3612 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/83a17a25cfd4775d64b748f0ec852d09a97f434b) will **not change** branch coverage
:no_entry: Merging #3612 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/83a17a25cfd4775d64b748f0ec852d09a97f434b) will will **increase** complexity by `29`


|           | [master](https://github.com/DataDog/dd-trace-dotnet/tree/83a17a25cfd4775d64b748f0ec852d09a97f434b) | #3612       | Change   | 
|:----------|:-----------:|:-----------:|:--------:|
| Lines     | `20927` / `29422` | `20963` / `29464` |          |
| Lines %   | `71%`      | `71%`      |  `0%` :heavy_check_mark:  |
| Branches  | `12562` / `18955` | `12564` / `18985` |          |
| Branches %| `66%`      | `66%`      |  `0%` :heavy_check_mark:  |
| Complexity|   `20454`      | `20483`        |  `29`  :no_entry:    |

View the full report for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=123096&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/123096/artifacts?artifactName=Code%20Coverage%20Report_123096&api-version=6.0&%24format=zip)  
* [HTML report for this PR #3612](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=123376&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/123376/artifacts?artifactName=Code%20Coverage%20Report_123376&api-version=6.0&%24format=zip)

### Datadog.Trace Breakdown :heavy_check_mark:

|        | [master](https://github.com/DataDog/dd-trace-dotnet/tree/83a17a25cfd4775d64b748f0ec852d09a97f434b) | #3612       | Change   | 
|:-------|:-----------:|:-----------:|:--------:|
| Lines %| `71%`      | `71%`       |  `0%` :heavy_check_mark:  |
| Branches %| `66%`      | `66%`       |  `0%` :heavy_check_mark:  |
| Complexity| `20454`      | `20483`       |  `29` :no_entry:  |

The following classes have significant coverage changes.

| File    | Line coverage change | Branch coverage change | Complexity change |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.Ci.Agent.Payloads.CICodeCoveragePayload](https://github.com/DataDog/dd-trace-dotnet/tree/7cd2b6d9a73869c30ca098c6ffd23d066375dc78/tracer/src/Datadog.Trace/Ci/Agent/Payloads/CICodeCoveragePayload.cs) | `-16%` :no_entry: | `0%` :heavy_check_mark:   | `1` :no_entry: |
| [Datadog.Trace.Ci.Agent.MessagePack.IEventMessagePackFormatter](https://github.com/DataDog/dd-trace-dotnet/tree/7cd2b6d9a73869c30ca098c6ffd23d066375dc78/tracer/src/Datadog.Trace/Ci/Agent/MessagePack/IEventMessagePackFormatter.cs) | `9%` :heavy_check_mark: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Ci.GitInfo](https://github.com/DataDog/dd-trace-dotnet/tree/7cd2b6d9a73869c30ca098c6ffd23d066375dc78/tracer/src/Datadog.Trace/Ci/GitInfo.cs) | `11%` :heavy_check_mark: | `8%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Ci.Agent.Payloads.MultipartPayload](https://github.com/DataDog/dd-trace-dotnet/tree/7cd2b6d9a73869c30ca098c6ffd23d066375dc78/tracer/src/Datadog.Trace/Ci/Agent/Payloads/MultipartPayload.cs) | `20%` :heavy_check_mark: | `38%` :heavy_check_mark:   | `-1` :heavy_check_mark: |
| [Datadog.Trace.Ci.TestSession](https://github.com/DataDog/dd-trace-dotnet/tree/7cd2b6d9a73869c30ca098c6ffd23d066375dc78/tracer/src/Datadog.Trace/Ci/TestSession.cs) | `40%` :heavy_check_mark: | `30%` :heavy_check_mark:   | `0` :heavy_check_mark: |


The following classes were added in #3612:

| File    | Line coverage | Branch coverage | Complexity |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.Ci.Agent.MessagePack.TestCoverageMessagePackFormatter](https://github.com/DataDog/dd-trace-dotnet/pull/3612/files) | `13%` | `0%` | `31` |


View the full reports for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=123096&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/123096/artifacts?artifactName=Code%20Coverage%20Report_123096&api-version=6.0&%24format=zip)  
* [HTML report for this PR #3612](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=123376&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/123376/artifacts?artifactName=Code%20Coverage%20Report_123376&api-version=6.0&%24format=zip)

**Review by OmerRaviv — APPROVED** [review]
LGTM

**Review by pierotibou — APPROVED** [review]
